tarantool/connection.py

```diff
@@ -1124,6 +1140,9 @@ def _recv(self, to_read):
         while to_read > 0:
             try:
                 tmp = self._socket.recv(to_read)
+            except BlockingIOError:
+                select.select([self._socket.fileno()], [], [])
```

Comment:
A `timeout` should be here.

Reply:
The behavior is the same as python does?

Reply:
Yes, I guess. From the `settimeout` method of socket library:

> If a non-zero value is given, subsequent socket operations will raise a [timeout](https://docs.python.org/3/library/socket.html#socket.timeout) exception if the timeout period value has elapsed before the operation has completed.

Reply:
> Yes, I guess. From the `settimeout` method of socket library:
> 
> > If a non-zero value is given, subsequent socket operations will raise a [timeout](https://docs.python.org/3/library/socket.html#socket.timeout) exception if the timeout period value has elapsed before the operation has completed.

exception socket.timeout

    A deprecated alias of [TimeoutError](https://docs.python.org/3/library/exceptions.html#TimeoutError).

It could be better to use TimeoutError.

Reply:
Changed to `TimeoutError`.